AnkiDroid/src/main/java/com/ichi2/preferences/StepsPreference.java

```diff
@@ -135,17 +136,17 @@ public static JSONArray convertToJSON(String steps) {
         }
         try {
             for (String s : steps.split("\\s+")) {
-                float f = Float.parseFloat(s);
+                double d = Double.parseDouble(s);
                 // 0 or less is not a valid step.
-                if (f <= 0) {
+                if (d <= 0) {
                     return null;
                 }
                 // Use whole numbers if we can (but still allow decimals)
-                int i = (int) f;
-                if (i == f) {
+                int i =(int) d;
```

Comment:
spacing nitpick, nothing more 
```suggestion
                int i = (int) d;
```

Reply:
Sure.
Screenshots:
![Floating_1](https://user-images.githubusercontent.com/62210041/120085518-1f9c0480-c0ee-11eb-820f-dc4bb508e0c6.jpg)

And the result is:

![Floating_2](https://user-images.githubusercontent.com/62210041/120085522-23c82200-c0ee-11eb-8153-81ba54345c98.jpg)